Modules/xASL_module_BIDS2Legacy.m

```diff
@@ -83,18 +80,20 @@
         % x.modules.bids2legacy.BIDS.subjects(1).perf(1) -> filename: 'sub-10015124_ses-1_run-1_asl.nii.gz'
         % x.modules.bids2legacy.BIDS.subjects(1).perf(2) -> filename: 'sub-10015124_ses-1_run-2_asl.nii.gz'
         % x.modules.bids2legacy.BIDS.subjects(1).perf(3) -> filename: 'sub-10015124_ses-1_run-1_m0scan.nii.gz'
-        % x.modules.bids2legacy.BIDS.subjects(1).perf(4) -> filename: 'sub-10015124_ses-1_run-2_m0scan.nii.gz'            
+        % x.modules.bids2legacy.BIDS.subjects(1).perf(4) -> filename: 'sub-10015124_ses-1_run-2_m0scan.nii.gz'  
+		%
+		% Also note that visits can now be any name, number, data, or a string and are always treated as a string
 
         % Subject ID
         SubjectID = x.modules.bids2legacy.BIDS.subjects(iSubjSess).name;
         SessionID = x.modules.bids2legacy.BIDS.subjects(iSubjSess).session;
         if isempty(SessionID)
-            iVisit = 1; % we default to a single visit
+            SessionID = '1';
         else
-            iVisit = str2num(SessionID(5:end));
+            SessionID = xASL_adm_CorrectName(SessionID(5:end), 2);
         end
 
-        SubjectVisit = [SubjectID '_' xASL_num2str(iVisit)]; % this is the legacy name
+        SubjectVisit = [SubjectID '_' SessionID]; % this is the legacy name
```

Comment:
SubjectVisit -> SubjectSession (we are in BIDS nomenclature)

Reply:
Done